### PR TITLE
Add Personality Database Presence

### PR DESCRIPTION
**Pull request by justarya**
website: https://www.personality-database.com/
Personality Database is the most popular site for discovering someone's personality type.

Website preview:
Screen Shot 2021-11-14 at 17 58 31

Screen Shot 2021-11-15 at 13 22 14

### CODE REVIEW

**Comment by EncryptedDev** [comment]
Why have you used setTrayTitle as the function is now deprecated? If you are wanting to still use it, please provide proof it works.

Also I suspect that you ran the linter on the entire presence folder, hence why you have the change for the other website, but if you are wanting to PR the change on 000webhost bump the version of it.

**Comment by justarya** [comment]
Here's proof of it. So far, works on my mac.

Screen Shot 2021-11-15 at 08 01 28

I have been scrolling through documentation and didn't find any other supported tray function.
Is there any function that can replace it?

**Comment by EncryptedDev** [comment]
> I have been scrolling through documentation and didn't find any other supported tray function.
Is there any function that can replace it?

No, the function was only intended for the purpose it was never used for, hence why it was deprecated

**Review by Slowlife01 — CHANGES_REQUESTED** [review]
Add few more things

**Comment by justarya** [comment]
## What's new
- Added new page support & Added new icon support

### Profile
Screen Shot 2021-11-19 at 21 53 40
Screen Shot 2021-11-19 at 21 48 23

### Searching
Screen Shot 2021-11-19 at 21 54 29
Screen Shot 2021-11-19 at 21 50 08

### Category
Screen Shot 2021-11-19 at 21 55 44
Screen Shot 2021-11-19 at 21 50 42

### Community Feed

Screen Shot 2021-11-19 at 21 56 26
Screen Shot 2021-11-19 at 21 50 59

### Notification
Screen Shot 2021-11-19 at 21 56 55
Screen Shot 2021-11-19 at 21 51 16

### User's profile
Screen Shot 2021-11-19 at 21 58 03

Screen Shot 2021-11-19 at 21 52 22
Screen Shot 2021-11-19 at 21 51 37

File: websites/P/Personality Database/presence.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * timestamp is started when the user opened the website
+   * and ended when the webiste is closed it
+   */
+  presenceData.startTimestamp = browsingStamp;
+
+  if (!presenceData.details) presence.setActivity();
+  else {
+    presence.setActivity(presenceData);
+  }
```
Comment (EncryptedDev) [inline websites/P/Personality Database/presence.ts:154]:
```suggestion
  else presence.setActivity(presenceData);
```

File: websites/P/Personality Database/presence.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    const title: string = document.title?.split(" | ")[0];
+    presenceData.details = "Viewing page:";
+    presenceData.state = title;
+  }
+
+  /**
+   * timestamp is started when the user opened the website
+   * and ended when the webiste is closed it
+   */
+  presenceData.startTimestamp = browsingStamp;
```
Comment (EncryptedDev) [inline websites/P/Personality Database/presence.ts:149]: Set this in the PresenceData object

File: websites/P/Personality Database/presence.ts [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        }
+      ];
+    } else presenceData.details = "Viewing user's profile";
+    presenceData.smallImageKey = "user";
+  } else {
+    /**
+     * show when viewing static or unrecognized pages
+     * ex: FAQ, Help, Community Guidelines
+     */
+    const title: string = document.title?.split(" | ")[0];
```
Comment (EncryptedDev) [inline websites/P/Personality Database/presence.ts:140]: Don't use a variable once #4893

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Could you also explain your rationale behind the exessive amounts of comments?

**Comment by justarya** [comment]
The comments are for context mostly, if others want to continue to develop this presence.

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**